### PR TITLE
refactor: add APM Otel native collector components

### PR DESCRIPTION
**Pull request by rogercoll**
Ready to review once the APM components available in `8.16-SNAPSHOT`

Configuration taken from: https://github.com/elastic/otel-apm-e2e-poc

### CODE REVIEW

**Comment by rogercoll** [comment]
It requires using these Helm Chart changes: https://github.com/open-telemetry/opentelemetry-helm-charts/pull/1384

File: resources/kubernetes/operator/helm/values.yaml [hunk 1]
```diff
@@ -184,17 +184,196 @@ collectors:
...
         debug:
           verbosity: basic
-        elasticsearch/apm:
-          endpoints:
-          - ${env:ELASTIC_ENDPOINT}
-          api_key: ${env:ELASTIC_API_KEY}
-          logs_dynamic_index:
-            enabled: true
-          mapping:
-            mode: ecs
```
Comment (gizas) [inline resources/kubernetes/operator/helm/values.yaml:197]: We remove the `elasticsearch/apm` and we are going to use elasticsearch/otel

Just clarifications: Dont  we need
```yaml
          traces_dynamic_index:
            enabled: true
          flush:
            interval: 10s
```

(meanwhile this config was used in elasticsearch/apm)

**Review by lahsivjar — COMMENTED** [review]
LGTM! Not approving as I am confirming if `event.success_count` is required as the current config doesn't handle this.

File: resources/kubernetes/operator/helm/values.yaml (hunk 1 above)
Comment (rogercoll) [reply resources/kubernetes/operator/helm/values.yaml:197]: good catch! It actually needs to be enabled, fixed https://github.com/elastic/opentelemetry/pull/27/commits/7c7941e671dbc0a1c3a4f1a864c0460e51e16456

**Comment by AlexanderWert** [comment]
> Not approving as I am confirming if event.success_count is required as the current config doesn't handle this.

I think we concluded that we need that, right? @lahsivjar could you propose a change for this on this PR?

File: resources/kubernetes/operator/helm/values.yaml [hunk 2]
```diff
@@ -184,23 +184,208 @@ collectors:
...
+              attributes:
+                - key: span.name
+                - key: event.outcome
+                - key: service.target.type
+                - key: service.target.name
+                - key: span.destination.service.resource
+                - key: metricset.name
+                  default_value: service_destination
+              sum:
+                value: Int(AdjustedCount())
```
Comment (lahsivjar) [inline resources/kubernetes/operator/helm/values.yaml:373]:
```suggestion
                value: Int(AdjustedCount())
      # event.success_count is populated using 2 metric definition with different conditions
      # and value for the histogram bucket based on event outcome. Both metric definition
      # are created using same name and attribute and will result in a single histogram.
      # We use mapping hint of aggregate_metric_double, so, only the sum and the count
      # values are required and the actual histogram bucket is ignored.
      - name: event.success_count
        description: Success count as a metric for service transaction
        include_resource_attributes:
          - key: service.name
          - key: deployment.environment # service.environment
          - key: telemetry.sdk.language # service.language.name
          - key: agent.name # set via elastictraceprocessor
        attributes:
          - key: transaction.root
          - key: transaction.type
          - key: metricset.name
            default_value: service_transaction
          - key: elasticsearch.mapping.hints
            default_value: [aggregate_metric_double]
        conditions:
          - attributes["event.outcome"] != nil and attributes["event.outcome"] == "success"
        unit: us
        histogram:
          buckets: [1]
          count: Int(AdjustedCount())
          value: Int(AdjustedCount())
      - name: event.success_count
        description: Success count as a metric for service transaction
        include_resource_attributes:
          - key: service.name
          - key: deployment.environment # service.environment
          - key: telemetry.sdk.language # service.language.name
          - key: agent.name # set via elastictraceprocessor
        attributes:
          - key: transaction.root
          - key: transaction.type
          - key: metricset.name
            default_value: service_transaction
          - key: elasticsearch.mapping.hints
            default_value: [aggregate_metric_double]
        conditions:
          - attributes["event.outcome"] != nil and attributes["event.outcome"] != "success"
        unit: us
        histogram:
          buckets: [0]
          count: Int(AdjustedCount())
          value: Double(0)
```

The above should add `event.success_count` metric.

**Review by gizas — APPROVED** [review]

**Review by lahsivjar — APPROVED** [review]

**Pull request merged**